### ISSUE TITLE
Remove redundant use of catch2

### ISSUE THREAD

**Issue by dthaler** [microsoft/ebpf-for-windows#3954]
### Describe the bug

catch2 is sycned twice:
external\catch2
external\usersim\external\catch2

The former can be removed and just use the latter.

### Steps taken to reproduce bug

Clone and sync recursively

### Expected behavior

catch2 should only be in one place

### Actual outcome

catch2 appears twice

**Comment by agarwalishita** [comment]
Can you please assign this issue to me? @dthaler

**Comment by agarwalishita** [comment]
Hey @dthaler , since it is my first contribution, I would really appreciate your guidance and any information you can provide on this issue.

**Comment by agarwalishita** [comment]
On cloning this repo, I found out that cache2 has been synced here:
Submodule 'external/Catch2' (https://github.com/catchorg/Catch2.git) registered for path 'external/Catch2'
Cloning into 'ebpf-for-windows/external/Catch2'...
Submodule path 'external/Catch2': checked out '8898cc61601af6cdd5b4548dd91ccf70ed67f3c2'

as well as

Submodule 'external/Catch2' (https://github.com/catchorg/Catch2.git) registered for path 'external/usersim/external/Catch2'
Cloning into 'ebpf-for-windows/external/usersim/external/Catch2'...
Submodule path 'external/usersim/external/Catch2': checked out 'fa43b77429ba76c462b1898d6cd2f2d7a9416b14'

Will work on removing the former one

**Comment by agarwalishita** [comment]
Hey @dthaler @shankarseal ,
Wouldn't it be better to remove Catch2 from the second location, since it is being consumed from the first location in most projects? This way, we avoid having to update paths everywhere. Removing it from the second path seems like a cleaner approach.

**Comment by dthaler** [comment]
Right, the latter (inside usersim) is needed because usersim has its own tests that use catch2, and it doesn't use other repositories that have catch2.
